YoCode/IDirectory.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+using System.IO;
+
+namespace YoCode
+{
+    public interface IDirectory
+    {
+        List<string> ModifiedPaths { get; }
```

Comment:
Same suggestion as before:
next to the Lists of original and modified paths, we should have two strings for their respective root directories